include/grpcpp/create_channel_posix.h

```diff
@@ -19,46 +19,34 @@
 #ifndef GRPCPP_CREATE_CHANNEL_POSIX_H
 #define GRPCPP_CREATE_CHANNEL_POSIX_H
 
-#include <memory>
-
-#include <grpc/support/port_platform.h>
-#include <grpcpp/channel.h>
-#include <grpcpp/support/channel_arguments.h>
+#include <grpcpp/create_channel_posix_impl.h>
 
 namespace grpc {
 
 #ifdef GPR_SUPPORT_CHANNELS_FROM_FD
```

Comment:
I never realized we had that. Gosh this is ugly. We should make a note to clean this up later.

Rule of thumb: no public API should be behind an ifdef like this. The actual implementation might differ depending on said define, and assert with a message saying this feature isn't enabled in this version, but the API/ABI shouldn't mutate depending on a compilation flag.

Reply:
CC @vjpai

Reply:
Filed https://github.com/grpc/grpc/issues/18571 to track.